### PR TITLE
LPS-182250 Avoid heap overload on DDMFieldUpgradeProccess

### PR DESCRIPTION
**Pull request by mateussgn**

### CODE REVIEW

**Comment by rafaprax** [comment]
ci:test:upgrade

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v4_1_0/DDMFieldUpgradeProcess.java [hunk 1]
```diff
@@ -96,65 +97,83 @@ protected void doUpgrade() throws Exception {
...
+					long companyId = resultSet.getLong(1);
+					long contentId = resultSet.getLong(2);
+					String data_ = resultSet.getString(3);
+					long structureId = resultSet.getLong(4);
+					long structureVersionId = resultSet.getLong(5);
+
+					return new Object[] {
+						contentId, companyId, data_, structureVersionId,
+						structureId
+					};
```
Comment (mateussgn) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v4_1_0/DDMFieldUpgradeProcess.java:127]: Should be:
```java
return new Object[] {
    contentId, companyId, data_, structured,
    structureVersionId
};
```

**Comment by rafaprax** [comment]
ci:test:sf

**Comment by rafaprax** [comment]
ci:test:upgrade

**Review by rafaprax — APPROVED** [review]

**Comment by mateussgn** [comment]
ci:test:upgrade

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v4_1_0/DDMFieldUpgradeProcess.java (hunk 1 above)
Comment (IstvanD) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v4_1_0/DDMFieldUpgradeProcess.java]: hi @mateussgn when I tried the upgrade process with this modification, it failed with the error
`com.liferay.portal.kernel.upgrade.UpgradeException: com.liferay.portal.kernel.upgrade.UpgradeException: Unable to find dynamic data mapping structure with ID 30850`
But without it, it was completed successfully.

Comment (mateussgn) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v4_1_0/DDMFieldUpgradeProcess.java]: Hi @IstvanD, 
I've made some other changes after this one, can you try again?
After I made the changes I tested with an upgrade from 7.1 to master and was completed successfully.

Comment (IstvanD) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v4_1_0/DDMFieldUpgradeProcess.java]: Hi @mateussgn thank you for the quick reply, I will test the changes. 👍 
I also got this error when upgrading the customer's database. Is this something that is maybe caused by the recent modifications?
```
2023-04-19 15:37:15.237 ERROR [main][UpgradeStepRegistratorTracker:185] Failed upgrade process for module com.liferay.dynamic.data.mapping.service
com.liferay.portal.kernel.upgrade.UpgradeException: java.sql.SQLSyntaxErrorException: Unknown column 'DDMStructure.structureVersionId' in 'field list'
```

Comment (IstvanD) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v4_1_0/DDMFieldUpgradeProcess.java]: @mateussgn are your most recent changes in the commit [3e7d309](https://github.com/mateussgn/liferay-portal/pull/92/commits/3e7d309df87d3d7b82b8854e0e00138a14702fdb) that was pushed two days ago? Thank you and best regards.

**Comment by mateussgn** [comment]
ci:close

**Pull request closed without merging**